Modules/Core/Common/include/itkConstNeighborhoodIterator.h

```diff
@@ -612,11 +612,14 @@ class ITK_TEMPLATE_EXPORT ConstNeighborhoodIterator
    *  within the buffer. */
   OffsetType m_WrapOffset{ { 0 } };
 
+  /** Default boundary condition. */
+  TBoundaryCondition m_InternalBoundaryCondition{};
+
   /** Pointer to the actual boundary condition that will be used.
    * By default this points to m_BoundaryCondition, but
    * OverrideBoundaryCondition allows a user to point this variable an external
    * boundary condition.  */
-  ImageBoundaryConditionPointerType m_BoundaryCondition{};
+  ImageBoundaryConditionPointerType m_BoundaryCondition{ &m_InternalBoundaryCondition };
```

Comment:
Is there some guarantee that `m_InternalBoundaryCondition` will be initialized before `m_BoundaryCondition`? Maybe because it was declared before? Or some other reason?

Reply:
Strictly speaking, I think it doesn't matter if m_InternalBoundaryCondition is already initialized. Even before it is initialized, its address `&m_InternalBoundaryCondition` should already be OK. 

I just moved the m_InternalBoundaryCondition declaration upward, before using `&m_InternalBoundaryCondition`, to avoid any confusion. But I don't think it really matters. I think it's just OK  😃 Thanks for asking!